Client/WebFilters/ContentBlocker/RequestBlockingContentHelper.swift

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import WebKit
+import BraveCore
+
+private let log = Logger.braveCoreLogger
+
+class RequestBlockingContentHelper: TabContentScript {
+  private struct RequestBlockingDTO: Decodable {
+    struct RequestBlockingDTOData: Decodable, Hashable {
+      enum CodingKeys: String, CodingKey {
+        case resourceURL, sourceURL, resourceType
+      }
+      
+      let resourceType: AdblockEngine.ResourceType
+      let resourceURL: URL
+      let sourceURL: URL
+      
+      init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let resourceURLString = try container.decode(String.self, forKey: .resourceURL)
+        let sourceURLString = try container.decode(String.self, forKey: .sourceURL)
+        let resourceTypeString = try container.decode(String.self, forKey: .resourceType)
```

Comment:
There is no need to decode as a string first, then create a `URL` from it.
You can do: `resourceURL = try container.decode(URL.self, forKey: .resourceURL)` for example.
It will decode `URL` which is `Codable` :)
This way, you can get rid of all the `DecodingError.dataCorrupted` stuff.

```swift
import Foundation

let json = """
{
  "url": "https://google.ca"
}
"""

struct Foo: Codable {
  let url: URL
}

let data = json.data(using: .utf8)!
let foo = try JSONDecoder().decode(Foo.self, from: data)

print(foo.url)
```

If you want to test in a playground.